internal/pkg/cli/app_init.go

```diff
@@ -202,15 +215,15 @@ func (o *initAppOpts) validateAppName(name string) error {
 	return nil
 }
 
-func (o *initAppOpts) validateDomain(domainName string) error {
-	domainExist, err := o.route53.DomainExists(domainName)
-	if err != nil {
-		return err
+func (o *initAppOpts) domainHostedZoneID(domainName string) (string, error) {
+	if o.cachedHostedZoneID != "" {
+		return o.cachedHostedZoneID, nil
 	}
-	if !domainExist {
-		return fmt.Errorf("no hosted zone found for %s", domainName)
+	hostedZoneID, err := o.route53.DomainHostedZoneID(domainName)
+	if err != nil {
+		return "", fmt.Errorf("get hosted zone ID for domain %s: %w", domainName, err)
 	}
-	return nil
+	return hostedZoneID, nil
```

Comment:
should we set `o.cachedHostedZoneID` here before the return? this way it's not only `Validate` that can cache it